packages/database/src/core/util/libs/parser.ts

```diff
@@ -174,17 +174,17 @@ export const parseDatabaseURL = function(
       colonInd = dataURL.length;
     }
 
-    const parts = host.split('.');
-    if (parts.length === 3) {
+    const hostWithoutPort = host.slice(0, colonInd);
```

Comment:
From the `slice` docs:

"A negative index can be used, indicating an offset from the end of the sequence. slice(2,-1) extracts the third element through the second-to-last element in the sequence."

So, in this case, if no port is specified it seems like we would drop the last character, no?

Reply:
If no port is specified, `colonInd = host.length` above, so `host.slice(0, host.length) ==> host`